src/schema/city/index.ts

```diff
@@ -100,21 +104,61 @@ export const City = {
   description: "A city-based entry point for local discovery",
   args: {
     slug: {
-      type: new GraphQLNonNull(GraphQLString),
+      type: GraphQLString,
       description:
         "A slug for the city, conforming to Gravity's city slug naming conventions",
     },
+    near: {
+      type: Near,
+      description:
+        "A point which will be used to locate the nearest local discovery city within a threshold",
+    },
   },
-  resolve: (_obj, args) => {
-    return lookupCity(args.slug)
+  resolve: (_obj, { slug, near }) => {
+    if (slug && near) {
+      throw new Error('The "slug" and "near" arguments are mutually exclusive.')
+    }
+    if (!slug && !near) {
+      throw new Error('One of the arguments "slug" or "near" is required.')
+    }
+    if (slug) {
+      return lookupCity(slug)
+    }
+    if (near) {
+      return nearestCity(near)
+    }
   },
 }
 
-const lookupCity = slug => {
+const lookupCity = (slug: string) => {
   if (!cityData.hasOwnProperty(slug)) {
     throw new Error(
       `City ${slug} not found in: ${Object.keys(cityData).join(", ")}`
     )
   }
   return cityData[slug]
 }
+
+const nearestCity = (latLng: LatLng) => {
+  const orderedCities: Point[] = citiesOrderedByDistance(latLng)
+  const closestCity: Point = orderedCities[0]
+
+  if (isCloseEnough(latLng, closestCity)) {
+    return closestCity
+  }
+  return null
+}
+
+const citiesOrderedByDistance = (latLng: LatLng): Point[] => {
+  let cities: Point[] = Object.values(cityData)
+  cities.sort((a: Point, b: Point) => {
```

Comment:
I don’t think you _need_ to type the function. I would assume `sort` is generic on the contents of the array and thus implicitly type the comparator parameters to be of the content type.